### PR TITLE
Base secubot on TALON UGV

### PR DESCRIPTION
**Pull request by ymber**
#### Summary
SUMMARY: Content "Rework secubot into a TALON UGV"

#### Purpose of change
Secubots are very similar to pre #33436 turrets. This brings them into line with existing systems that serve their function.

#### Describe the solution
- Add M16A4, including appropriate itemgroups, recipes, and gunmod compatibility lists for use by player
- Remove unused `secubot` itemgroup
- Replace secubot MP5 with M16A4
- Reduce secubot maximum ammo to 30 rounds
- Reduce secubot speed to 40
- Increase secubot range to 30
- Add 200 move targeting time and targeting warning to secubot
- Set secubot to semi automatic fire only

#### Additional context
The SWORDS weapon mount they use on TALON UGVs can take other weapons, I just picked the M16 because it's the lightest and easiest of them. You can mount M249s and M240s and apparently M107s and the M202 FLASH. We could get some good enemy variety out of these.

### CODE REVIEW

**Comment by tenmillimaster** [comment]
Who gave Johnny 5 a gun.

**Comment by LaVeyanFiend** [comment]
> The SWORDS weapon mount they use on TALON UGVs can take other weapons, I just picked the M16 because it's the lightest and easiest of them. You can mount M249s and M240s and apparently M107s and the M202 FLASH. We could get some good enemy variety out of these.

Source?

**Comment by Firestorm01X2** [comment]
>Replace secubot MP5 with M16A4
>Increase secubot range to 30

What is your rationale of increasing range and damage of the bots? How much dangerous you expect them to be for player?

**Comment by ymber** [comment]
@LaVeyanFiend There's [an article](https://archive.defense.gov/transformation/articles/2004-12/ta120604c.html) in the DOD's archive and [this paper](https://apps.dtic.mil/dtic/tr/fulltext/u2/a478652.pdf).

@Firestorm01X2 Increased damage and range comes from how we model ballistics. The SWORDS mount can't actually take an MP5 and this change bring the monster into line with reality so I have it a rifle that would work. These changes should make secubots less dangerous in most circumstances. They're substantially slower now and they always made a lot of noise when they move, and in combination with the 200 move targeting time this should make them easier and safer to hunt and they'll have a harder time catching you out as long as you're maneuvering. They're more dangerous then they used to be if they're fighting in in ideal circumstances for them but as long as you don't give them ideal circumstances they shouldn't be that hard to handle. All the old tactics against secubots will still work except trying to stay out of their range, and most of them will work better with slower secubots with a slower targeting time.

File: data/json/monsters/defense_bot.json [hunk 1]
```diff
@@ -20,8 +20,27 @@
     "armor_cut": 14,
     "vision_day": 50,
     "revert_to_itype": "bot_secubot",
-    "starting_ammo": { "9mm": 100 },
-    "special_attacks": [ { "type": "gun", "cooldown": 2, "gun_type": "hk_mp5", "ranges": [ [ 0, 6, "AUTO" ], [ 7, 12, "DEFAULT" ] ] } ],
+    "starting_ammo": { "223": 30 },
+    "special_attacks": [
```
Comment (kevingranade) [inline data/json/monsters/defense_bot.json]:
```suggestion
    "special_attacks": [
```
Trailing space.

File: data/json/monsters/defense_bot.json [hunk 2]
```diff
@@ -20,8 +20,27 @@
...
+        "require_targeting_npc": true,
+        "require_targeting_monster": true,
+        "laser_lock": false,
+        "targeting_cost": 200,
+        "targeting_timeout_extend": -10,
+        "targeting_sound": "\"Hostile detected.\"",
+        "targeting_volume": 50,
+        "no_ammo_sound": "a chk!"
+      }
+     ],
```
Comment (kevingranade) [inline data/json/monsters/defense_bot.json]:
```suggestion
    ],
```
Indention.

**Pull request merged**